packages/blockchain-link-utils/src/solana.ts

```diff
@@ -0,0 +1,398 @@
+                            acc[token.contract].balance || '0',
+                        )
+                            .plus(token.balance || '0')
+                            .toString();
+                        acc[token.contract].accounts!.push({
+                            publicKey: token.address,
+                            balance: token.balance || '0',
+                        });
+                    } else {
+                        const { type, contract, balance, decimals, name, symbol } = token;
+                        acc[token.contract] = {
+                            type,
+                            contract,
+                            balance,
+                            decimals,
+                            name,
+                            symbol,
+                            accounts: [{ publicKey: token.address, balance: balance || '0' }],
+                        };
+                    }
+
+                    return acc;
+                },
+            ),
+            D.values,
+        ),
+    );
+
+    return tokens;
+};
+
+// First step in parsing a tx, is getting Solana effects on accounts that were in the transaction, from this effects we later parse the other tx properties.
+export const extractAccountBalanceDiff = (
+    transaction: ParsedTransactionWithMeta,
+    address: string,
+    isTokenDiff = false,
+): {
+    preBalance: BigNumber;
+    postBalance: BigNumber;
+} | null => {
+    const pubKeyIndex = transaction.transaction.message.accountKeys.findIndex(
+        ak => ak.pubkey.toString() === address,
+    );
+
+    if (pubKeyIndex === -1) {
+        return null;
+    }
+
+    if (isTokenDiff) {
+        const preBalance = transaction.meta?.preTokenBalances?.find(
+            balance => balance.accountIndex === pubKeyIndex,
+        )?.uiTokenAmount.amount;
+        const postBalance = transaction.meta?.postTokenBalances?.find(
+            balance => balance.accountIndex === pubKeyIndex,
+        )?.uiTokenAmount.amount;
+
+        return {
+            preBalance: new BigNumber(preBalance ?? 0),
+            postBalance: new BigNumber(postBalance ?? 0),
+        };
+    }
+
+    const preBalance = transaction.meta?.preBalances[pubKeyIndex];
+
+    const postBalance = transaction.meta?.postBalances[pubKeyIndex];
+
+    return {
+        preBalance: new BigNumber(preBalance ?? 0),
+        postBalance: new BigNumber(postBalance ?? 0),
+    };
+};
+
+const isWSolTransfer = (ixs: (ParsedInstruction | PartiallyDecodedInstruction)[]) =>
+    ixs.find(
+        ix =>
+            'parsed' in ix &&
+            ix.parsed.info?.mint === 'So11111111111111111111111111111111111111112',
+    );
+
+type TransactionEffect = {
+    address: string;
+    amount: BigNumber;
+};
+
+export function getNativeEffects(transaction: ParsedTransactionWithMeta): TransactionEffect[] {
+    // TODO(vl): after token PR is merged, get WSOL mint from there
```

Comment:
could be that this Todo is outdated now?

Reply:
Yep, fixed.